x-pack/filebeat/input/cel/input.go

```diff
@@ -541,6 +541,25 @@ func (i input) run(env v2.Context, src *source, cursor map[string]interface{}, p
 						pubCursor = cursor
 					}
 				}
+				// This is checked prior to the publish attempt since the
+				// cursor.Publisher interface does not document the behaviour
+				// related to context cancellation and the context is not
+				// explicitly passed in, so favour this explicit clarity.
+				switch err := ctx.Err(); {
+				case err == nil:
+				case errors.Is(err, context.Canceled), errors.Is(err, context.DeadlineExceeded):
+					log.Infow("context cancelled with unpublished events", "unpublished", len(events)-i)
+					// Don't update status, since we are about to pass
+					// through the Running state and then fall through
+					// to the input exit with a change to Stopped.
+					break
+				default:
+					// This should never happen.
+					log.Warnw("failed with unpublished events", "error", err, "unpublished", len(events)-i)
+					health.UpdateStatus(status.Degraded, "error publishing events: "+err.Error())
+					isDegraded = true
+					break
```

Comment:
I had one question here, in the default case it signals something critical or unexpected, so why are we breaking and going back to sleep instead of returning and stopping the input as a whole ?